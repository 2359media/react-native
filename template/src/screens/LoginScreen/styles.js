import {createStylesheet} from '../../utils/styleUtils';

const styles = createStylesheet({
  input: {
    fontSize: 16,
    height: 48,
    padding: 8,
    borderRadius: 8,
    borderWidth: 1,
    borderColor: '#EEE',
    marginBottom: 16,
  },
});

export default styles;
