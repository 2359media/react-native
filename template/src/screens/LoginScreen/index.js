import React, {useState} from 'react';
import {TextInput} from 'react-native';
import {useDispatch} from 'react-redux';
import {useRequest} from '../../utils/apiUtils';
import AppAction from '../../actions/AppAction';
import AccountApi from '../../apis/AccountApi';
import Screen from '../../components/Screen';
import TopNavigation from '../../components/TopNavigation';
import ContentView from '../../components/ContentView';
import Button from '../../components/Button';
import styles from './styles';

const LoginScreen = ({navigation}) => {
  const [username, setUsername] = useState('john');
  const [password, setPassword] = useState('1234');
  const request = useRequest();
  const dispatch = useDispatch();

  function login() {
    request(AccountApi.login(username, password), {
      showLoading: true,
      showError: true,
    }).then(d => {
      dispatch(AppAction.setToken(d.data.token));
      navigation.reset({index: 0, routes: [{name: 'Home'}]});
    });
  }

  return (
    <Screen>
      <TopNavigation title="Login" />
      <ContentView>
        <TextInput
          placeholder="username"
          style={styles.input}
          textContentType="username"
          testID="userInput"
          value={username}
          onChangeText={setUsername}
        />
        <TextInput
          placeholder="password"
          style={styles.input}
          textContentType="password"
          secureTextEntry
          testID="passInput"
          value={password}
          onChangeText={setPassword}
        />
        <Button testID="loginButton" text="Login" onPress={login} />
      </ContentView>
    </Screen>
  );
};

export default LoginScreen;
