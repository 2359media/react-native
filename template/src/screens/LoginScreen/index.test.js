import React from 'react';
import {
  renderer,
  TestContainer,
  navigation,
  Alert,
} from '../../utils/testUtils';
import LoginScreen from '.';

const tree = renderer(<TestContainer Component={LoginScreen} />);

test('snapshot', () => {
  expect(tree).toMatchSnapshot();
});

test('success login', () => {
  tree.run('userInput', 'onChangeText', 'john');
  tree.run('passInput', 'onChangeText', '1234');
  tree.run('loginButton');
  expect(navigation.reset).toBeCalledWith({
    index: 0,
    routes: [{name: 'Home'}],
  });
});

test('failed login', () => {
  tree.run('passInput', 'onChangeText', '4321');
  tree.run('loginButton');
  expect(navigation.reset).not.toBeCalled();
  expect(Alert.alert).toBeCalledWith('Error', 'Wrong username or password');
});
