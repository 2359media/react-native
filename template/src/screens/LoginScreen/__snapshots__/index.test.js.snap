// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "elevation": 5,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "shadowColor": "black",
          "shadowOpacity": 0.2,
          "shadowRadius": 5,
          "zIndex": 100,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 60,
          "paddingHorizontal": 16,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "minWidth": 50,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 8,
              "left": 8,
              "right": 8,
              "top": 8,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "margin": 4,
                "tintColor": "#EC632B",
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "alignSelf": "stretch",
            "flex": 1,
            "justifyContent": "center",
            "paddingVertical": 8,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "black",
              "fontSize": 20,
              "fontWeight": "bold",
              "left": 50,
              "position": "absolute",
              "right": 50,
              "textAlign": "center",
            }
          }
        >
          Login
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "minWidth": 50,
          }
        }
      />
    </View>
  </View>
  <RCTScrollView
    contentContainerStyle={
      Array [
        Object {
          "flexGrow": 1,
          "paddingBottom": 24,
          "paddingLeft": 24,
          "paddingRight": 24,
          "paddingTop": 24,
        },
        undefined,
      ]
    }
    keyExtractor={[Function]}
    keyboardShouldPersistTaps="handled"
  >
    <View>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="username"
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#EEE",
            "borderRadius": 8,
            "borderWidth": 1,
            "fontSize": 16,
            "height": 48,
            "marginBottom": 16,
            "padding": 8,
          }
        }
        testID="userInput"
        textContentType="username"
        underlineColorAndroid="transparent"
        value="john"
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="password"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "borderColor": "#EEE",
            "borderRadius": 8,
            "borderWidth": 1,
            "fontSize": 16,
            "height": 48,
            "marginBottom": 16,
            "padding": 8,
          }
        }
        testID="passInput"
        textContentType="password"
        underlineColorAndroid="transparent"
        value="1234"
      />
      <View
        accessible={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 8,
            "left": 8,
            "right": 8,
            "top": 8,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#EC632B",
            "borderRadius": 8,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "padding": 8,
          }
        }
        testID="loginButton"
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "bold",
              "margin": 4,
            }
          }
        >
          Login
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
